### PR TITLE
AuthenticationController, AuthenticationService: login return JWT token

### PR DESCRIPTION
**Pull request by BohdanS-UA**
JwtAuthenticationFilter, JwtUtil, OpenApiConfig, UserLoginResponseDto added

### CODE REVIEW

File: src/main/java/mate/academy/bookstorev2/config/OpenApiConfig.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class OpenApiConfig {
+    @Bean
+    public OpenAPI customOpenApi() {
+        return new OpenAPI()
+                .components(new Components().addSecuritySchemes("BearerAuth",
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstorev2/config/OpenApiConfig.java]: Create constants for String BearerAuth and etc

File: src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            UserDetails userDetails = userDetailsService.loadUserByUsername(username);
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java]: Use String constant for Authorization from HttpHeaders

File: src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer")) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java]: create constant for "Bearer " (don't forget whitespace)

File: src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java]: use constant above and length() instead of 7

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            UserDetails userDetails = userDetailsService.loadUserByUsername(username);
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(AUTHORIZATION);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    public static final String AUTHORIZATION = "Authorization";
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    public static final String BEARER = "Bearer";
+    public static final int TOKEN_INDEX = 7;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    public static final String AUTHORIZATION = "Authorization";
+    public static final String BEARER = "Bearer";
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java]:
```suggestion
    public static final String TOKEN_HEADER = "Bearer ";
```

File: src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(AUTHORIZATION);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(BEARER)) {
+            return bearerToken.substring(TOKEN_INDEX);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstorev2/security/JwtAuthenticationFilter.java]:
```suggestion
        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(TOKEN_HEADER)) {
            return bearerToken.substring(TOKEN_HEADER.length());
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by okuzan — APPROVED** [review]

**Pull request merged**